x/wasm/keeper/ante.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"github.com/CosmWasm/wasmd/x/wasm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type CountTXDecorator struct {
+	storeKey sdk.StoreKey
+}
+
+func NewCountTXDecorator(storeKey sdk.StoreKey) *CountTXDecorator {
+	return &CountTXDecorator{storeKey: storeKey}
+}
+
+func (a CountTXDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if simulate {
+		return next(types.WithTXCounter(ctx, 0), tx, simulate)
+	}
+	currentHeight := ctx.BlockHeight()
+	store := ctx.KVStore(a.storeKey)
+	var txCounter uint64 = 0
```

Comment:
🤔  I tried to follow the advice but there is also the case where the store is empty that requires a `0` init value. I am not sure if it would get easier by adding `else` blocks. Instead I have added more code comments.